decryptor/mysql/prepared_statements.go

```diff
@@ -281,3 +285,103 @@ func (m *mysqlBoundValue) Encode() (encoded []byte, err error) {
 
 	return encoded, outErr
 }
+
+// PreparedStatementFieldTracker track and replace DataType for column and param ColumnDefinition
+type PreparedStatementFieldTracker struct {
+	proxyHandler *Handler
+	// shared value that indicates number of param packet
+	paramsCounter int
+	columnsNum    uint16
+}
+
+// NewPreparedStatementFieldTracker create new PreparedStatementFieldTracker
+func NewPreparedStatementFieldTracker(handler *Handler, columnNum uint16) PreparedStatementFieldTracker {
+	return PreparedStatementFieldTracker{
+		proxyHandler: handler,
+		columnsNum:   columnNum,
+	}
+}
+
+// ParamsTrackHandler ResponseHandler used to track prepared statement params
+func (p *PreparedStatementFieldTracker) ParamsTrackHandler(ctx context.Context, packet *Packet, _, clientConnection net.Conn) error {
```

Comment:
what about to test it at least via unit-tests? we can in unit tests pass our implementations of `net.Conn` which will be just a `bytes.Buffer`. Then call `ParamsTrackHandler` with settings that expects replacing types. After that read output, parse explicitly and verify that type was changed. Via `ParseResultField` function. Or refactor this function to be able pass just `data []byte` value, not `packet`.

Reply:
Yes, will copy example packet from Wireshark:)